### PR TITLE
Test multiple storageclasses data deletion with compression and replication

### PR DESCRIPTION
**Pull request by shyRozen**
Add test to verify that data from pools that have replica 2,3 and compression is deleted after IO and pvc delete.

### CODE REVIEW

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+        for pod_obj in pod_obj_list:
+            get_fio_rw_iops(pod_obj)
+
+        log.info("deleting PODs and PVCs")
+        delete_pods(pod_obj_list, wait=True)
+        delete_pvcs(pvc_obj_list, concurrent=True)
+
+        log.info("Wait for 5 seconds for all data to delete")
+        sleep(15)
```
Comment (Avilir) [inline tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:88]: can you fix one of the values - the log said 5 sec. but you are sleeping for 15 sec.

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        log.info("Checking stats after deleting PODs and PVCs")
+        for sc_obj in sc_obj_list:
+            pvc_list = get_all_pvcs_in_storageclass(sc_obj.name)
+            if len(pvc_list) == 0:
+                cbp_name = sc_obj.get()["parameters"]["pool"]
+                ceph_pool_byte_used = get_byte_used_by_pool(cbp_name)
+                log.info(f"pool {cbp_name} has {ceph_pool_byte_used} bytes used")
+                if ceph_pool_byte_used > 250000:
```
Comment (Avilir) [inline tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:96]: why you are comparing with this number (250000) and not with 0 ?

Comment (shyRozen) [reply tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:96]: There is leftovers of ~200KB left in the pool. Verified with engineers that this is legit. This is some metadata that stays on the pool and gets cleaned later. Most of the data which is in gigs get deleted.

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py (hunk 1 above)
Comment (shyRozen) [reply tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:88]: Thanks! Fixed

File: ocs_ci/ocs/cluster.py [hunk 3]
```diff
@@ -954,6 +954,28 @@ def validate_replica_data(pool_name, replica):
...
+
+    Returns:
+        integer: The amount of byte stored from pool.
+
+    """
+    ceph_df_detail_output = get_ceph_df_detail()
+    pool_list = ceph_df_detail_output.get("pools")
+    for pool in pool_list:
+        if pool.get("name") == pool_name:
+            logger.info(f"{pool_name}")
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:973]: Logging the pool name only doesn't tell much. I suggest logging the pool name along with the byte used.

File: ocs_ci/ocs/cluster.py [hunk 4]
```diff
@@ -954,6 +954,28 @@ def validate_replica_data(pool_name, replica):
...
+            byte_used = pool["stats"]["bytes_used"]
+            return byte_used
+    raise PoolNotFound(f"Pool {pool_name} not found on cluster")
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:976]: It's better to return a value and process it at the test level so if needed, the exception will be raised there and not here

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    PoolDataNotErased,
+    PvcNotDeleted,
+)
+from ocs_ci.ocs.cluster import get_byte_used_by_pool
+from ocs_ci.ocs.resources.pod import delete_pods
+from ocs_ci.ocs.resources.pvc import (
+    delete_pvcs,
+    get_all_pvcs_in_storageclass,
+)
+from time import sleep
```
Comment (ebenahar) [inline tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:20]: this should be above the ocs_ci import statements, at line 2

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+log = logging.getLogger(__name__)
+
+
+@tier1
```
Comment (ebenahar) [inline tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:25]: this test should be part of tier2

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        This test function does below,
+        *. Creates 2 Storage Class with creating new rbd pool
+        *. Creates PVCs using new Storage Class
+        *. Mount PVC to an app pod
+        *. Run IO on an app pod
+        *. Delete the pods and pvc
+        *. Verify that the data is deleted
+
+        """
+        log.info("Creating storageclasses")
```
Comment (ebenahar) [inline tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:50]: please elaborate more on the newly created storage class

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        interface_type = constants.CEPHBLOCKPOOL
+        sc_obj1 = storageclass_factory(
+            interface=interface_type,
+            new_rbd_pool=True,
+            replica=3,
+            compression="aggressive",
+        )
+
```
Comment (ebenahar) [inline tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:58]: add logging for the 2nd storage class creation

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        log.info("Wait for 15 seconds for all data to delete")
+        sleep(15)
+        log.info("Checking stats after deleting PODs and PVCs")
+        for sc_obj in sc_obj_list:
+            pvc_list = get_all_pvcs_in_storageclass(sc_obj.name)
+            if len(pvc_list) == 0:
+                cbp_name = sc_obj.get()["parameters"]["pool"]
+                ceph_pool_byte_used = get_byte_used_by_pool(cbp_name)
+                log.info(f"pool {cbp_name} has {ceph_pool_byte_used} bytes used")
+                if ceph_pool_byte_used > 250000:
```
Comment (ebenahar) [inline tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:96]: I suggest saving 250000 as a default in ocs_ci/ocs/defaults.py

File: ocs_ci/ocs/cluster.py [hunk 10]
```diff
@@ -954,6 +954,28 @@ def validate_replica_data(pool_name, replica):
     raise PoolNotFound(f"Pool {pool_name} not found on cluster")
 
 
+def get_byte_used_by_pool(pool_name):
+    """
+
```
[the remark this answers is not in the document]
Comment (shyRozen) [reply ocs_ci/ocs/cluster.py:959]: Deleted

File: ocs_ci/ocs/cluster.py (hunk 3 above)
Comment (shyRozen) [reply ocs_ci/ocs/cluster.py:973]: This is leftovers from debugging. The pool and the bytes are logged in the test itselft
```log.info(f"pool {cbp_name} has {ceph_pool_byte_used} bytes used")```
I'll delete the pool logged here

File: ocs_ci/ocs/cluster.py (hunk 4 above)
Comment (shyRozen) [reply ocs_ci/ocs/cluster.py:976]: The return value is processed at test level. This is an exception related if the pool not found at all. and the recommendation to do it this way came from Martin in another PR. Exception regarding return value is raised at the test level.

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py (hunk 5 above)
Comment (shyRozen) [reply tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:20]: Done

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py (hunk 9 above)
Comment (shyRozen) [reply tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py:96]: done

File: tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py (hunk 6 above)
Comment (shyRozen) [reply tests/manage/storageclass/test_multiple_sc_comp_rep_data_delete.py]: done

File: ocs_ci/ocs/cluster.py (hunk 4 above)
Comment (ebenahar) [reply ocs_ci/ocs/cluster.py:974]: understood now. Let's keep it as is

**Review by ebenahar — APPROVED** [review]

**Review by Avilir — APPROVED** [review]

**Pull request merged**